website/content/docs/agent/autoauth/methods/oci.mdx

```diff
@@ -0,0 +1,43 @@
+---
+layout: docs
+page_title: Vault Agent Auto-Auth OCI (Oracle Cloud Infrastructure) Method
+description: OCI (Oracle Cloud Infrastructure) Method for Vault Agent Auto-Auth
+---
+
+# Vault Agent Auto-Auth OCI (Oracle Cloud Infrastructure) Method
+
+The `oci` method performs authentication against the [OCI Auth
+method](/vault/docs/auth/oci).
+
+## Credentials
+
+The method use to authenticate is set using the `type` parameter. Valid values are `apikey` to authenticate using
+API Key credentials and `instance` for Instance Principal credentials.
+
+If `apikey` is used, the Vault agent will use the first credential it can successfully obtain in the following order:
+
+1. Environment variables:
+  - `OCI_tenancy_ocid`
+  - `OCI_user_ocid`
+  - `OCI_fingerprint`
+  - `OCI_private_key_path`
+2. Configuration file in `$HOME/.oci/config`
+3. Path to configuration file defined in the `OCI_CONFIG_FILE` environment variable
+4. Configuration file in `$HOME/.obmcs/config`
+
+Wherever possible, we recommend using instance principal for credentials. These are rotated automatically by OCI
+and require no effort on your part to provision, making instance principal the most secure of the three methods. If
+using instance principal _and_ a custom `credential_poll_interval`, be sure the frequency is set to a value that is less
+than OCI's rotation frequency. This is currently documented as
+[multiple times a day](https://docs.oracle.com/en-us/iaas/Content/Identity/Tasks/callingservicesfrominstances.htm#faq),
+but from experience, credentials are rotated every 10 to 15 minutes.
+
+## Configuration
+
+### General
+
+- `type` `(string: required)` - The type of authentication to use. Valid values are `apikey` and `instance`.
+
+- `role` `(string: required)` - The role to authenticate against on Vault.
+
+- `credential_poll_interval` `(integer: optional)` - In seconds, how frequently the Vault agent should check for new credentials.
```

Comment:
This should be a duration string instead: https://developer.hashicorp.com/vault/docs/concepts/duration-format, as these are the standard that HashiCorp uses for intervals like this.

These can be parsed using `parseutil.ParseDurationSecond()`, and can be documented like this:

`(duration: "60s", optional)`  - In seconds, how frequently the Vault agent should check for new credentials. Uses [duration format strings](/docs/concepts/duration-format).`

Reply:
The AWS and Alicloud implementations also use `(integer: optional)`, so I also used integer for consistency:
- https://github.com/hashicorp/vault/blob/main/website/content/docs/agent/autoauth/methods/aws.mdx#configuration
- https://github.com/hashicorp/vault/blob/main/website/content/docs/agent/autoauth/methods/alicloud.mdx#configuration
If you still prefer `duration`, I am happy to change it as well.

Reply:
I see! That makes sense, but I think let's still go with duration. That's the standard around Vault for everything new, and it does still work with integers.

Reply:
Resolved in b50cf06